vmm/src/vm.rs

```diff
@@ -1446,8 +1446,7 @@ pub fn test_vm() {
     let load_addr = GuestAddress(0x1000);
     let mem = GuestMemoryMmap::from_ranges(&[(load_addr, mem_size)]).unwrap();
 
-    let kvm = hypervisor::kvm::KvmHypervisor::new().unwrap();
-    let hv: Arc<dyn hypervisor::HypervisorTrait> = Arc::new(kvm);
+    let hv: Arc<dyn hypervisor::HypervisorTrait> = Arc::new(hypervisor::Hypervisor::new().unwrap());
```

Comment:
No I wouldn't proceed this way. Instead, we should have two different hypervisor initializations gated by a feature. Both would return a structure implementing the Hypervisor trait.

Reply:
To be clear: do you want these two different hypervisor initializations be put in the `vmm` crate or the `hypervisor` crate?

Reply:
The hypervisor crate should be agnostic from initialization and how things are being used. That's why I would suggest the two different hypervisor initialization would be put in the `vmm` crate. They would be gated by feature.